### PR TITLE
[🍒 stable/20240723] Add logs for SymbolFileDWARF::FindTypes

### PR DESCRIPTION
**Pull request by DmT021**
Cherry-pick https://github.com/llvm/llvm-project/pull/106030

`SymbolFileDWARF::FindTypes` was logged prior to [this commit](https://github.com/llvm/llvm-project/commit/dd9587795811ba21e6ca6ad52b4531e17e6babd6#diff-edef3a65d5d569bbb75a4158d35b827aa5d42ee03ccd3b0c1d4f354afa12210c). This is a helpful log message for checking for redundant type searches

### CODE REVIEW

**Comment by DmT021** [comment]
@augusto2112 PTAL

**Comment by augusto2112** [comment]
@swift-ci test

**Review by augusto2112 — APPROVED** [review]
Awesome! Thanks!

**Comment by augusto2112** [comment]
@DmT021 we will have to wait until the linux bots are fixed to merge this

**Comment by augusto2112** [comment]
@swift-ci test

**Pull request merged**
